### PR TITLE
Move session logic to Session Ops

### PR DESCRIPTION
**Pull request by emma-p**
Also:
- adds tests for the session logic
- moves the redirect logic and error display into private controller methods
- renames csrf package to auth package

### CODE REVIEW

File: app/controllers/Login.scala [hunk 1]
```diff
@@ -43,22 +44,26 @@ object Login extends Controller with AuthActions {
...
-          session + (UserIdentity.KEY -> Json.toJson(identity).toString) - ANTI_FORGERY_KEY - LOGIN_ORIGIN_KEY
-        }
+        val redirect = successfulLoginRedirect(session)
+        redirect.withSession { session.loggedIn(identity, LOGIN_ORIGIN_KEY) }
       }
       case _ => {
-        Redirect(loginCall).withSession(session - ANTI_FORGERY_KEY)
-          .flashing("error" -> Messages("groups.failure", GoogleGroups.requiredGroups.mkString(", ")))
+        val redirect = loginErrorRedirect("groups.failure", GoogleGroups.requiredGroupsMsg)
+        redirect.withSession { session.loginError }
```
Comment (markjamesbutler) [inline app/controllers/Login.scala]: Nice implicit conversion. ;)

File: app/auth/CSRFAction.scala [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package csrf
+package auth
```
Comment (markjamesbutler) [inline app/auth/CSRFAction.scala]: What was the reason for this package move?

Comment (emma-p) [inline app/auth/CSRFAction.scala]: Now it also contains the LoginSession so I changed the name to something more generic than csrf, but maybe they need to live in different packages?

Comment (markjamesbutler) [inline app/auth/CSRFAction.scala]: Ok, I'll review it when looking at the CSRF code so good for now.

**Comment by markjamesbutler** [comment]
LGTM :+1:

**Pull request merged**